core/src/main/scala/org/apache/spark/scheduler/ResultTask.scala

```diff
@@ -65,4 +65,6 @@ private[spark] class ResultTask[T, U](
   override def preferredLocations: Seq[TaskLocation] = preferredLocs
 
   override def toString = "ResultTask(" + stageId + ", " + partitionId + ")"
+
+  override def canEqual(other: Any): Boolean = other.isInstanceOf[ResultTask[T, U]]
```

Comment:
eh...StageId and partitionId is like a unique composite primary key in database.  In current spark context, it's sure can be identified by (StageId, PartitionId), even not need to use "canEqual".